include/nbl/video/surface/ISurfaceGL.h

```diff
@@ -77,12 +85,30 @@ class CSurfaceGLImpl final : public Base<Window>
 
 template <typename Window>
 using CSurfaceGL = CSurfaceGLImpl<Window, CSurface>;
-template <typename Window>
-using CSurfaceNativeGL = CSurfaceGLImpl<Window, CSurfaceNative>;
+template <typename Window, typename CRTP>
+using CSurfaceNativeGL = CSurfaceGLImpl<Window, CSurfaceNative, CRTP>;
 
 // TODO: conditional defines
 using CSurfaceGLWin32 = CSurfaceGL<ui::IWindowWin32>;
-using CSurfaceNativeGLWin32 = CSurfaceNativeGL<ui::IWindowWin32>;
+class CSurfaceNativeGLWin32 : public CSurfaceNativeGL<ui::IWindowWin32, CSurfaceNativeGLWin32>
+{
+protected:
+    using base_t = CSurfaceNativeGL<ui::IWindowWin32, CSurfaceNativeGLWin32>;
+    using base_t::base_t;
+
+    uint32_t getWidth() const override 
+    { 
+        RECT wr;
+        GetWindowRect(m_handle, &wr);
+        return wr.right - wr.left;
+    }
+    uint32_t getHeight() const override 
+    { 
+        RECT wr;
+        GetWindowRect(m_handle, &wr);
+        return wr.top - wr.bottom;
+    }
```

Comment:
who did this? @achalpandeyy @Erfan-Ahmadi ?

what's wrong with the window size the `IWindowWin32` would report?

Reply:
me
for surfaces created from native window handles
Needed for baw

Reply:
aah ok, can it break because of threading / out of sync stuff?

Reply:
whenever you call `GetWindowRect` you're calling it from whatever thread calls `ISurfaceGL::getWidth` etc. while all the window stuff gets done on the window manager thread?